modules/ejbca-ejb/src/org/ejbca/core/protocol/cmp/RevocationMessageHandler.java

```diff
@@ -234,40 +234,40 @@ public ResponseMessage handleMessage(final BaseCmpMessage msg, boolean authentic
             final String errMsg = INTRES.getLocalizedMessage("cmp.errormissingissuerrevoke",
                     (issuer != null ? issuer.toString() : "<no issuer in request>"),
                     (serno != null ? serno.getValue().toString(16) : "<no serial number in request>"));
-		    failText = errMsg; 
+		    failText = errMsg;
 		    LOG.info(failText);
 		}
-		
+
 		if (LOG.isDebugEnabled()) {
 		    LOG.debug("Creating a PKI revocation message response: "+responseProtection);
 		}
 		final CmpRevokeResponseMessage rresp = new CmpRevokeResponseMessage();
 		rresp.setRecipientNonce(msg.getSenderNonce());
 		rresp.setSenderNonce(new String(Base64.encode(CmpMessageHelper.createSenderNonce())));
-		// The revocation message may have had an empty recipient, in which case we got the recipient from the CMP configuration (see above)
-		if (StringUtils.isEmpty(msg.getRecipient().getName().toString())) {
-		    final X509Certificate cacert = (X509Certificate)ca.getCACertificate();
-		    final GeneralName sender = new GeneralName(X500Name.getInstance(cacert.getSubjectX500Principal().getEncoded()));
-		    rresp.setSender(sender);
-		} else {
-	        rresp.setSender(msg.getRecipient());		    
-		}
 		rresp.setRecipient(msg.getSender());
 		rresp.setTransactionId(msg.getTransactionId());
 		rresp.setFailInfo(failInfo);
 		rresp.setFailText(failText);
 		rresp.setStatus(status);
 
 		if (StringUtils.equals(responseProtection, "pbe")) {
-		    final HMACAuthenticationModule hmacmodule = (HMACAuthenticationModule) authenticationModule;
+			// The revocation message may have had an empty recipient, in which case we got the recipient from the CMP configuration (see above)
```

Comment:
It would be great to add references to RFCs, as noted in the github discussion. Just a line or two of java comments. It's helpful for long term maintenance.

Reply:
Done